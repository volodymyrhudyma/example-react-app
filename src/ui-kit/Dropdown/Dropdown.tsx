import React from 'react';
import { Wrapper, DropdownButton } from './styled/Dropdown.styled';
import OutsideClickHandler from 'react-outside-click-handler';

interface IDropdownProps {
  close: () => void;
  action: () => void;
  buttonText: string;
}

const Dropdown = ({ close, action, buttonText }: IDropdownProps) => (
  <Wrapper>
    <OutsideClickHandler onOutsideClick={close}>
      <DropdownButton onClick={action}>{buttonText}</DropdownButton>
    </OutsideClickHandler>
  </Wrapper>
);

export default Dropdown;
