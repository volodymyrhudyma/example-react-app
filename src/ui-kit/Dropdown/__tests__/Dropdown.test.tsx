import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from '../Dropdown';

describe('Dropdown component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Dropdown
        close={jest.fn()}
        action={jest.fn()}
        buttonText="Mark All as Read"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
