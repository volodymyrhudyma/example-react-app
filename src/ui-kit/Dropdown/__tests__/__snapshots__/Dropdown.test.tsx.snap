// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render correctly 1`] = `
<styled.div>
  <OutsideClickHandler
    disabled={false}
    display="block"
    onOutsideClick={[MockFunction]}
    useCapture={true}
  >
    <styled.div
      onClick={[MockFunction]}
    >
      Mark All as Read
    </styled.div>
  </OutsideClickHandler>
</styled.div>
`;
